utils/generate-commands-json.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+# Generate a commands.json file from the output of the `COMMAND` command
+# Prerequisites: Python 3.5+ and redis-py
+import argparse
+import functools
+import json
+import redis
+
+def convert_list_to_dict(arr, recurse):
+    rep = {}
+    for i in range(0,len(arr),2):
+        key = arr[i].replace('-', '_')
+        val = arr[i+1]
+        if type(val) is list and recurse:
+            val = convert_list_to_dict(val, recurse)
+        rep[key] = val
+    return rep
+
+def convert_flags_to_truthy_dict(flags):
+    rep = {}
+    for f in flags:
+        rep[f] = True
+    return rep
+
+def convert_argument(arg):
+    rep = convert_list_to_dict(arg, False)
+    if 'flags' in rep:
+        rep = {
+            **rep,
+            **convert_flags_to_truthy_dict(rep['flags'])
+        }
+        del rep['flags']
+    if 'arguments' in rep:
+        rep['arguments'] = list(map(lambda x: convert_argument(x), rep['arguments']))
```

Comment:
why not
```
rep['arguments'] = [convert_argument(x) for x in rep['arguments']]
```

a bit more readble IMHO

Reply:
Yeah, I forgot all my list comprehension